script/verification/SystemConfigUpgradeEcotoneScalars.s.sol

```diff
@@ -0,0 +1,48 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.15;
+
+import {console2 as console} from "forge-std/console2.sol";
+import {Vm} from "forge-std/Vm.sol";
+import {LibString} from "solady/utils/LibString.sol";
+import {SuperchainRegistry} from "script/verification/Verification.s.sol";
+import "@eth-optimism-bedrock/src/dispute/lib/Types.sol";
+import {ISystemConfig} from "./ISystemConfig.sol";
+import {IResourceMetering} from "./IResourceMetering.sol";
+import {MIPS} from "@eth-optimism-bedrock/src/cannon/MIPS.sol";
+import {Simulation} from "@base-contracts/script/universal/Simulation.sol";
+import {NestedMultisigBuilder} from "@base-contracts/script/universal/NestedMultisigBuilder.sol";
+import {SystemConfigUpgrade} from "script/verification/SystemConfigUpgrade.s.sol";
+
+contract SystemConfigUpgradeEcotoneScalars is SystemConfigUpgrade {
+    using LibString for string;
+
+    constructor(string memory l1ChainName, string memory l2ChainName, string memory release)
+        SystemConfigUpgrade(l1ChainName, l2ChainName, release)
+    {}
+
+    /// @notice Public function that must be called by the verification script.
+    function checkSystemConfigUpgrade() public view override {
+        ISystemConfig sysCfg = ISystemConfig(proxies.SystemConfig);
+        uint256 reencodedScalar =
+            (uint256(0x01) << 248) | (uint256(sysCfg.blobbasefeeScalar()) << 32) | sysCfg.basefeeScalar();
```

Comment:
Is this only needed temporarily since we are upgrading the SystemConfig to a version that no longer needs encoding?

Reply:
The `sysCfg.blobbasefeeScalar()` and `sysCfg.basefeeScalar()` getters read from dedicated storage slots. When we did the upgrade incorrectly previously, these storage slots were not updated causing the getters to return `0`. Now, we are checking they return values consistent with what you get if you call `sysCfg.scalar()` and decode them by hand. 

Essentially we implicitly construct the expected values for the two new getters by reencoding what we get after the upgrade and checking it matches the previous `scalar` value.

I could try redesigning this so it is more explicit, and decode the expected values from the previous.scalar and compare the new getter's results with those?